### PR TITLE
Replace DrawableRuleset class with an interface

### PR DESCRIPTION
**Pull request by frenzibyte**
- Replaces abstract class `DrawableRuleset` with a drawable interface `IDrawableRuleset` as told by the xmldoc remarks.

### CODE REVIEW

**Comment by peppy** [comment]
What's the point / reasoning of this change?

File: osu.Game/Screens/Play/Player.cs [hunk 1]
```diff
@@ -161,9 +161,9 @@ private void addGameplayComponents(Container target, WorkingBeatmap working)
                 .WithChild(beatmapSkinProvider
                     .WithChild(target = rulesetSkinProvider)));
 
-            target.AddRange(new Drawable[]
+            target.AddRange(new[]
             {
-                DrawableRuleset,
+                DrawableRuleset as Drawable,
```
Comment (peppy) [inline osu.Game/Screens/Play/Player.cs]: this is the reason for the xmldoc that you removed. casting an interface to a drawable like this is super-dangerous.

Comment (iiSaLMaN) [reply osu.Game/Screens/Play/Player.cs]: but it's an `IDrawable`, it can't be inherited from a normal class, no?

**Comment by iiSaLMaN** [comment]
> What's the point / reasoning of this change?

Just for code quality, I saw it while reading on the drawable ruleset implementation and thought why not do that because `IDrawable` already exists

File: osu.Game/Screens/Play/Player.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Screens/Play/Player.cs]: there's nothing in place to enforce as such, which is why we weren't using an interface until now. if you can explain what this change is for (ie. what is dependent on switching to an interface) that may help figure a correct direction here.

**Comment by peppy** [comment]
On a quick git history check, `IDrawable` has always existed (since 2017) but that comment was added in 2019. Until `IDrawable` can be used in place of `Drawable` (without casts) switching to using it for cases like this is not feasible.

**Comment by iiSaLMaN** [comment]
Closing this PR until `IDrawable` is used instead of `Drawable`

**Pull request closed without merging**